include/fs/littlefs.h

```diff
@@ -33,12 +38,28 @@ struct fs_littlefs {
 	 */
 	uint32_t *lookahead_buffer[CONFIG_FS_LITTLEFS_LOOKAHEAD_SIZE / sizeof(uint32_t)];
 
+	/* If not set will be set to use the flash operation */
+	lfs_device_open device_open;
+
+	/* If not set will be set to use the flash operation */
+	lfs_device_close device_close;
+
+	/* If not set will be set to use the flash operation */
+	lfs_device_get_block_size device_get_block_size;
+
+	/* If not set will be set to use the flash operation */
+	lfs_device_get_block_size device_get_size;
+
```

Comment:
Maybe for the purpose define structure like `device_services` that will gather all the callbacks, and add pointer to such structure here. Because you are removing pointer to `flash_area`, adding pointer here will keep the size of `fs_littlefs` structure the same and will allow to reuse `device_services` among different mount points.

Reply:
done